app/controllers/search_controller.rb

```diff
@@ -121,4 +133,76 @@ def format_record_type(record_type, obsolete = false)
     record_text
   end
 
-end
+  def group_results_by_ontology(results)
+    results.group_by { |element| element['links']['ontology'] }
+  end
+
+  def make_search_result(grouped_results)
+    search_results = []
+    grouped_results.each_key do |key|
+      element_pref_lab = get_closest_preflab(grouped_results[key][0].prefLabel) 
+      element_id = grouped_results[key][0].id
```

Comment:
extract vaibel `grouped_results[key].first`, instead of calling ` grouped_results[key][0]`each time